scripts/check_branch.py

```diff
@@ -72,11 +72,14 @@ def current_branch():
     try:
         # Gitlab checks out a specific commit which results in a detached HEAD
         # (no active branch). Extract the branch name from the runner environment.
-        branch = os.environ['CI_COMMIT_REF_NAME']
+        branch_name = os.environ['CI_COMMIT_REF_NAME']
     except KeyError:
+        # Detached head may also occur outside of the gitlab environment, in
+        # which case it is only allowed for personal deployments.
         repo = git.Repo(config.project_root)
-        branch = repo.active_branch.name
-    return branch
+        branch_name = 'DETACHED HEAD' if repo.head.is_detached else repo.active_branch.name
```

Comment:
Before inventing special magic values, you should always consider if the type system can be used to express special cases. A detached head implies that there is no branch checked out so `None` is much more adequate value, even if it requires more changes downstream of this function.

Reply:
I chose this value to produce informative error messages.

Reply:
I understand. I didn't mean to imply that you chose this value without consideration. My version also produces informative messages but doesn't have the flaw of overloading the branch name when there really is no branch.